### PR TITLE
build: fix windows CI by updating subprojects/xz and subprojects/libzip

### PR DESCRIPTION
**Pull request by ret2libc**
Hopefully it fixes the problem with appveyor

### CODE REVIEW

**Comment by ret2libc** [comment]
Oops, it's intrin.h, not immintrin.h

**Comment by XVilka** [comment]
@ret2libc see https://github.com/tukaani-project/xz/issues/48#issuecomment-1514859469

File: subprojects/packagefiles/xz-f41df2ac2fed347d3f107f3533e76e000d29c6cb/config.h.meson [hunk 1]
```diff
@@ -498,3 +498,7 @@
 /* Define to the type of an unsigned integer type wide enough to hold a
    pointer, if such a type exists, and if the system does not define it. */
 #mesondefine uintptr_t
+
```
Comment (XVilka) [inline subprojects/packagefiles/xz-f41df2ac2fed347d3f107f3533e76e000d29c6cb/config.h.meson]: This probably could be removed.

**Review by XVilka — CHANGES_REQUESTED** [review]
It fixed the xz problem, now there is one with libzip:
```
./subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(45,5): error: incompatible function pointer types initializing 'HANDLE (*)(const void *, DWORD, DWORD, PSECURITY_ATTRIBUTES, DWORD, DWORD, HANDLE) __attribute__((stdcall))' (aka 'void *(*)(const void *, unsigned long, unsigned long, struct _SECURITY_ATTRIBUTES *, unsigned long, unsigned long, void *)') with an expression of type 'HANDLE (const char *, DWORD, DWORD, PSECURITY_ATTRIBUTES, DWORD, DWORD, HANDLE) __attribute__((stdcall))' (aka 'void *(const char *, unsigned long, unsigned long, struct _SECURITY_ATTRIBUTES *, unsigned long, unsigned long, void *)') [-Wincompatible-function-pointer-types]
    utf16_create_file,
    ^~~~~~~~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(46,5): error: incompatible function pointer types initializing 'BOOL (*)(const void *) __attribute__((stdcall))' (aka 'int (*)(const void *)') with an expression of type 'BOOL (LPCWSTR) __attribute__((stdcall))' (aka 'int (const unsigned short *)') [-Wincompatible-function-pointer-types]
    DeleteFileW,
    ^~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(47,5): error: incompatible function pointer types initializing 'DWORD (*)(const void *) __attribute__((stdcall))' (aka 'unsigned long (*)(const void *)') with an expression of type 'DWORD (LPCWSTR) __attribute__((stdcall))' (aka 'unsigned long (const unsigned short *)') [-Wincompatible-function-pointer-types]
    GetFileAttributesW,
    ^~~~~~~~~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(48,5): error: incompatible function pointer types initializing 'BOOL (*)(const void *, GET_FILEEX_INFO_LEVELS, void *) __attribute__((stdcall))' (aka 'int (*)(const void *, enum _GET_FILEEX_INFO_LEVELS, void *)') with an expression of type 'BOOL (LPCWSTR, GET_FILEEX_INFO_LEVELS, LPVOID) __attribute__((stdcall))' (aka 'int (const unsigned short *, enum _GET_FILEEX_INFO_LEVELS, void *)') [-Wincompatible-function-pointer-types]
    GetFileAttributesExW,
    ^~~~~~~~~~~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(50,5): error: incompatible function pointer types initializing 'BOOL (*)(const void *, const void *, DWORD) __attribute__((stdcall))' (aka 'int (*)(const void *, const void *, unsigned long)') with an expression of type 'BOOL (LPCWSTR, LPCWSTR, DWORD) __attribute__((stdcall))' (aka 'int (const unsigned short *, const unsigned short *, unsigned long)') [-Wincompatible-function-pointer-types]
    MoveFileExW,
    ^~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(51,5): error: incompatible function pointer types initializing 'BOOL (*)(const void *, DWORD) __attribute__((stdcall))' (aka 'int (*)(const void *, unsigned long)') with an expression of type 'BOOL (LPCWSTR, DWORD) __attribute__((stdcall))' (aka 'int (const unsigned short *, unsigned long)') [-Wincompatible-function-pointer-types]
    SetFileAttributesW,
    ^~~~~~~~~~~~~~~~~~
```

Opened a bug in libzip repo: https://github.com/nih-at/libzip/issues/375

**EDITED**: I found also this PR that was merged upstream: https://github.com/nih-at/libzip/pull/322

**Comment by XVilka** [comment]
@ret2libc you can use "master" branch for `xz` now, it was merged and will be a part of the next release.

**Comment by XVilka** [comment]
@ret2libc you can use xz-5.4.3 now:
- https://github.com/tukaani-project/xz/releases/tag/v5.4.3

**Comment by wargio** [comment]
```
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(45,5): error: incompatible function pointer types initializing 'HANDLE (*)(const void *, DWORD, DWORD, PSECURITY_ATTRIBUTES, DWORD, DWORD, HANDLE) __attribute__((stdcall))' (aka 'void *(*)(const void *, unsigned long, unsigned long, struct _SECURITY_ATTRIBUTES *, unsigned long, unsigned long, void *)') with an expression of type 'HANDLE (const char *, DWORD, DWORD, PSECURITY_ATTRIBUTES, DWORD, DWORD, HANDLE) __attribute__((stdcall))' (aka 'void *(const char *, unsigned long, unsigned long, struct _SECURITY_ATTRIBUTES *, unsigned long, unsigned long, void *)') [-Wincompatible-function-pointer-types]
    utf16_create_file,
    ^~~~~~~~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(46,5): error: incompatible function pointer types initializing 'BOOL (*)(const void *) __attribute__((stdcall))' (aka 'int (*)(const void *)') with an expression of type 'BOOL (LPCWSTR) __attribute__((stdcall))' (aka 'int (const unsigned short *)') [-Wincompatible-function-pointer-types]
    DeleteFileW,
    ^~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(47,5): error: incompatible function pointer types initializing 'DWORD (*)(const void *) __attribute__((stdcall))' (aka 'unsigned long (*)(const void *)') with an expression of type 'DWORD (LPCWSTR) __attribute__((stdcall))' (aka 'unsigned long (const unsigned short *)') [-Wincompatible-function-pointer-types]
    GetFileAttributesW,
    ^~~~~~~~~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(48,5): error: incompatible function pointer types initializing 'BOOL (*)(const void *, GET_FILEEX_INFO_LEVELS, void *) __attribute__((stdcall))' (aka 'int (*)(const void *, enum _GET_FILEEX_INFO_LEVELS, void *)') with an expression of type 'BOOL (LPCWSTR, GET_FILEEX_INFO_LEVELS, LPVOID) __attribute__((stdcall))' (aka 'int (const unsigned short *, enum _GET_FILEEX_INFO_LEVELS, void *)') [-Wincompatible-function-pointer-types]
    GetFileAttributesExW,
    ^~~~~~~~~~~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(50,5): error: incompatible function pointer types initializing 'BOOL (*)(const void *, const void *, DWORD) __attribute__((stdcall))' (aka 'int (*)(const void *, const void *, unsigned long)') with an expression of type 'BOOL (LPCWSTR, LPCWSTR, DWORD) __attribute__((stdcall))' (aka 'int (const unsigned short *, const unsigned short *, unsigned long)') [-Wincompatible-function-pointer-types]
    MoveFileExW,
    ^~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_utf16.c(51,5): error: incompatible function pointer types initializing 'BOOL (*)(const void *, DWORD) __attribute__((stdcall))' (aka 'int (*)(const void *, unsigned long)') with an expression of type 'BOOL (LPCWSTR, DWORD) __attribute__((stdcall))' (aka 'int (const unsigned short *, unsigned long)') [-Wincompatible-function-pointer-types]
    SetFileAttributesW,
    ^~~~~~~~~~~~~~~~~~
```
Please add a cast directly in a patch

**Comment by XVilka** [comment]
```
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_ansi.c(43,5): error: incompatible function pointer types initializing 'create_file_td' (aka 'void *(*)(const void *, unsigned long, unsigned long, struct _SECURITY_ATTRIBUTES *, unsigned long, unsigned long, void *)') with an expression of type 'HANDLE (LPCSTR, DWORD, DWORD, LPSECURITY_ATTRIBUTES, DWORD, DWORD, HANDLE) __attribute__((stdcall))' (aka 'void *(const char *, unsigned long, unsigned long, struct _SECURITY_ATTRIBUTES *, unsigned long, unsigned long, void *)') [-Wincompatible-function-pointer-types]
    CreateFileA,
    ^~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_ansi.c(44,5): error: incompatible function pointer types initializing 'delete_file_td' (aka 'int (*)(const void *)') with an expression of type 'BOOL (LPCSTR) __attribute__((stdcall))' (aka 'int (const char *)') [-Wincompatible-function-pointer-types]
    DeleteFileA,
    ^~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_ansi.c(45,5): error: incompatible function pointer types initializing 'get_file_attributes_td' (aka 'unsigned long (*)(const void *)') with an expression of type 'DWORD (LPCSTR) __attribute__((stdcall))' (aka 'unsigned long (const char *)') [-Wincompatible-function-pointer-types]
    GetFileAttributesA,
    ^~~~~~~~~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_ansi.c(46,5): error: incompatible function pointer types initializing 'get_file_attributes_ex_td' (aka 'int (*)(const void *, enum _GET_FILEEX_INFO_LEVELS, void *)') with an expression of type 'BOOL (LPCSTR, GET_FILEEX_INFO_LEVELS, LPVOID) __attribute__((stdcall))' (aka 'int (const char *, enum _GET_FILEEX_INFO_LEVELS, void *)') [-Wincompatible-function-pointer-types]
    GetFileAttributesExA,
    ^~~~~~~~~~~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_ansi.c(48,5): error: incompatible function pointer types initializing 'move_file_td' (aka 'int (*)(const void *, const void *, unsigned long)') with an expression of type 'BOOL (LPCSTR, LPCSTR, DWORD) __attribute__((stdcall))' (aka 'int (const char *, const char *, unsigned long)') [-Wincompatible-function-pointer-types]
    MoveFileExA,
    ^~~~~~~~~~~
../subprojects/libzip-1.9.2/lib/zip_source_file_win32_ansi.c(49,5): error: incompatible function pointer types initializing 'set_file_attributes_td' (aka 'int (*)(const void *, unsigned long)') with an expression of type 'BOOL (LPCSTR, DWORD) __attribute__((stdcall))' (aka 'int (const char *, unsigned long)') [-Wincompatible-function-pointer-types]
    SetFileAttributesA,
    ^~~~~~~~~~~~~~~~~~
```

**Review by XVilka — APPROVED** [review]
xz and libzip problems are solved. Though I hope libzip addresses the problem in their code eventually. The one problem that still left is:
```
../librz/util/thread.c(22,9): error: incompatible pointer to integer conversion returning 'void *' from a function with result type 'DWORD' (aka 'unsigned long') [-Wint-conversion]
        return NULL;
               ^~~~
C:\Program Files\LLVM\lib\clang\16\include\stddef.h(89,16): note: expanded from macro 'NULL'
#  define NULL ((void*)0)
               ^~~~~~~~~~
1 error generated.
[379/1978] Compiling C object librz/util/rz_util-0.6.dll.p/table.c.obj
[380/1978] Compiling C object librz/util/rz_util-0.6.dll.p/thread_cond.c.obj
```
I think it can be addressed separately, thus merging this PR as is.

**Comment by wargio** [comment]
https://github.com/rizinorg/rizin/pull/3504

**Pull request merged**